File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/exception/UnsuccessResponseException.java
package com.happs.ximand.ringcontrol.model.object.exception;

import com.happs.ximand.ringcontrol.R;

public class UnsuccessResponseException extends BluetoothException {

    private final int code;

    public UnsuccessResponseException(int code) {
        this.code = code;
    }

    @Override
    public int getMessageResId() {
        return R.string.unsuccess_response;
    }

    @Override
    public int getCode() {
        return code;
    }
}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/exception/UnsuccessResponseException.kt
package com.happs.ximand.ringcontrol.model.`object`.exception

import com.happs.ximand.ringcontrol.R

class UnsuccessResponseException(code: Int) : BaseException(code) {

    override val descriptionResId = R.string.unsuccessfully_response_description

    override val showSupportLayout = true

}